speedy-recovery-client/src/components/secured/home/HomePageForParent.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { Loader } from "semantic-ui-react";
+
+class HomePageForParent extends Component {
+  render() {
+    if (
+      this.props.user &&
+      this.props.events &&
+      this.getNextEvent(this.props.events).start &&
+      this.getNextEvent(this.props.events).patient
+    ) {
+      return (
+        <div>
+          <p>Hello {this.props.user.name}</p>
+          {new Date(this.getNextEvent(this.props.events).start).toLocaleString(
```

Comment:
Please don't do a big conditional test like this in-line. Instead, use a helper function like
```
function isValidDate(d) {
  return d instanceof Date && !isNaN(d);
}
```
Also, just write this once (probably in one of the services) and call it from all 3 types of HomePage.